File: src/main/java/miragefairy2019/mod/modules/ore/material/ItemBlockMaterials.java
package miragefairy2019.mod.modules.ore.material;

import miragefairy2019.mod.lib.multi.ItemBlockMulti;
import net.minecraft.item.ItemStack;

public class ItemBlockMaterials<V extends IBlockVariantMaterials> extends ItemBlockMulti<BlockMaterials<V>, V> {

    public ItemBlockMaterials(BlockMaterials<V> block) {
        super(block);
    }

    @Override
    public int getItemBurnTime(ItemStack itemStack) {
        return block2.getVariant(itemStack.getMetadata()).getBurnTime();
    }

}

File: src/main/java/miragefairy2019/mod/modules/ore/material/ItemBlockMaterials.kt
package miragefairy2019.mod.modules.ore.material

import miragefairy2019.mod.lib.multi.ItemBlockMulti
import net.minecraft.item.ItemStack

class ItemBlockMaterials<V : IBlockVariantMaterials>(block: BlockMaterials<V>) : ItemBlockMulti<BlockMaterials<V>, V>(block) {
    override fun getItemBurnTime(itemStack: ItemStack) = block2.getVariant(itemStack.metadata).burnTime
}
